orchagent/neighorch.cpp

```diff
@@ -402,17 +425,61 @@ bool NeighOrch::removeNextHop(const IpAddress &ipAddress, const string &alias)
 
     assert(hasNextHop(nexthop));
 
+    SWSS_LOG_INFO("Removing next hop %s", nexthop.toLogString().c_str());
+
     gFgNhgOrch->invalidNextHopInNextHopGroup(nexthop);
 
     if (m_syncdNextHops[nexthop].ref_count > 0)
     {
-        SWSS_LOG_ERROR("Failed to remove still referenced next hop %s on %s",
-                       ipAddress.to_string().c_str(), alias.c_str());
+        SWSS_LOG_ERROR("Failed to remove still referenced next hop %s",
+                       nexthop.toLogString().c_str());
         return false;
     }
 
+    sai_object_id_t next_hop_id = m_syncdNextHops[nexthop].next_hop_id;
+    sai_status_t status = sai_next_hop_api->remove_next_hop(next_hop_id);
+
+    /*
+     * If the next hop removal fails and not because the next hop doesn't
+     * exist, return false.
+     */
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        /* When next hop is not found, we continue to remove neighbor entry. */
+        if (status == SAI_STATUS_ITEM_NOT_FOUND)
+        {
+            SWSS_LOG_ERROR("Failed to locate next hop %s, rv:%d",
+                           nexthop.toLogString().c_str(), status);
+        }
```

Comment:
@prsunny Should the CRM resource be decremented when the NHOP Object was not found in SDK.